packages/untp-test-suite/src/core/services/dynamic-loading-schemas/utils.ts

```diff
@@ -0,0 +1,48 @@
+import fs from 'fs';
+import path from 'path';
+import { fileURLToPath } from 'url';
+
+/**
+ * Check if a file exists
+ * @param filePath - The path to the file
+ * @returns A promise that resolves to a boolean indicating if the file exists
+ */
+const checkFileExists = async (filePath: string): Promise<boolean> => {
+  try {
+    await fs.promises.access(filePath);
+    return true; // The file exists
+  } catch {
+    return false; // The file does not exist
+  }
+};
+const __dirname = path.dirname(fileURLToPath(import.meta.url));
+const SCHEMA_PATH = path.resolve(__dirname, '../../../schemas'); // ../tests-untp/packages/untp-test-suite/src/schemas
+
+/**
+ * Check if a schema exists
+ * @param schemaName - The name of the schema
+ * @returns A promise that resolves to a boolean indicating if the schema exists
+ */
+export const checkSchemaExists = async (schemaName: string): Promise<boolean> => {
+  return checkFileExists(`${SCHEMA_PATH}/${schemaName}`);
+};
+
+/**
+ * Check if a schema version exists
+ * @param folderName - The name of the schema
+ * @param version - The version of the schema
+ * @returns A promise that resolves to a boolean indicating if the schema version exists
+ */
+export const checkSchemaVersionExists = async (folderName: string, version: string): Promise<boolean> => {
+  return checkFileExists(`${SCHEMA_PATH}/${folderName}/${version}`);
+};
+
+/**
+ * Get the content of a schema
+ * @param folderName - The name of the schema
+ * @param version - The version of the schema
+ * @returns A promise that resolves to the content of the schema
+ */
+export const getSchemaContent = (folderName: string, version: string): Promise<string> => {
+  return fs.promises.readFile(`${SCHEMA_PATH}/${folderName}/${version}/schema.json`, 'utf-8');
+};
```

Comment:
I think when we use the `fs.readFile` function, it handles checking whether the file exists or not. If the file does not exist at the specified path, it will throw an error for us. This error is very clear, allowing the developer who is implementing to easily understand the issue. Therefore, we don't need to manually check whether every subfolder contains the `schema.json` file.

Reply:
> I think when we use the `fs.readFile` function, it handles checking whether the file exists or not. If the file does not exist at the specified path, it will throw an error for us. This error is very clear, allowing the developer who is implementing to easily understand the issue. Therefore, we don't need to manually check whether every subfolder contains the `schema.json` file.

We want to wrap up the loading file for schemas to a separating service, then the structure is quite clear and easy to change in the future. The wrapper here is also returning the specific message to easy understand